_api-reference/ingest-apis/processors/append.md

```diff
@@ -0,0 +1,100 @@
+---
+layout: default
+title: Append
+parent: Ingest processors 
+grand_parent: Ingest APIs
+nav_order: 10
+---
+
+# Append
+
+The `append` processor is used to add values to a field:
+- If the field is an array, the `append` processor appends the specified values to that array.
+- If the field is a scalar field, the `append` processor converts it to an array and appends the specified values to that array.
+- If the field does not exist, the `append` processor creates an array with the specified values.
+
+The syntax for the `append` processor is: 
+
+```json
+{
+    "append": {
+        "field": "field_name",
+        "value": ["value1", "value2", "{{value3}}"]
+    }
+}
+```
+
+## Parameters
+
+The following table lists the required and optional parameters for the `append` processor.
+
+**Parameter** | **Required** | **Description** |
+|-----------|-----------|-----------|
+`field`  | Required  | Name of the field where the data should be appended. Supports template snippets.|
+`value`  | Required  | Value to be appended. This can be a static value, a dynamic value derived from existing fields, or a value obtained from external lookups. Supports template snippets. |
+`allow_duplicates`  | Optional  | If set to `false`, the processor will not append values that already exist in the field. Default is `true`.  |
+`description`  | Optional  | Brief description of the processor.  |  
+`if` | Optional | Condition to execute this processor. |
+`on_failure` | Optional | A list of processors to execute if the processor fails. |
+`ignore_failure` | Optional | If set to `true`, failures are ignored. Default is `false`. |
+`tag` | Optional | An identifier tag for the processor. Useful for debugging to distinguish between processors of the same type. |
```

Comment:
@heemin32 Will you provide me an example of the tag? I'm getting errors when I test. Thank you.

Reply:
```
{
  "processors" : [
    {
      "trim": {
        "field": "user",
	"tag": "my test processor"
      }
    }
  ]
}
```